### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by isaacpitwa**
## Bookstore: add reducers and actions
- Add React Redux 
- Structure your application files using a "feature folder" approach and use the [ducks pattern](https://github.com/erikras/ducks-modular-redux) for your Redux files. Your files/folder structure could look like this:
- Write your book's actions and reducer (in /src/redux/books/books.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).
- Define action types for adding and removing a book.
      - Set the initial state to be an empty array of books.
      - Export Action Creators for your actions.
      - Write your reducer and export it as default.
           - Define state changes for the actions that you created.
           - In case of unknown action - return to the current state.
- Write your categories actions and reducer (in /src/redux/categories/categories.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).
- Define an action type for checking the status.
        - Set the initial state to be an empty array of categories.
        - Export Action Creators for your actions.
        - Write your reducer and export it as default.
             - For the check status action return a string "Under construction".
             - In case of unknown action - return the current state.Configure the Redux Store (/src/redux/configureStore.js):

-  Importing the necessary methods from Redux.
       - Combine both reducers into a root reducer by using combineReducers function.
       - Use the root reducer to create your store.
- fixed linter errors

### CODE REVIEW

File: src/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    type: REMOVE_BOOK,
+    book,
+  };
+}
+
+export default function reducer(state = [{
+  id: '1',
+  title: 'Test book one',
+  author: 'Test Author',
+}], action = {}) {
```
Comment (codepantha) [inline src/redux/books/books.js:22]: - [x] For your reducer, please remove the test data and set an initial state with an empty array 🙏
- [ ] Through the action, you get access to the type that was dispatched and also the payload. So action should not be set to an empty object. Please fix this 😎

File: src/redux/configureStore.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { combineReducers } from 'redux';
+import { configureStore } from '@reduxjs/toolkit';
```
Comment (codepantha) [inline src/redux/configureStore.js:2]: - [ ] You're currently importing `configureStore` from `@reduxjs/toolkit;` however, in this milestone, you are expected to 
import `createStore` from `redux`. Please make this change 👍

**Review by codepantha — CHANGES_REQUESTED** [review]
Hi @isaacpitwa 👋,

## STATUS: REQUESTED CHANGES ♻️  

Good job so far 👏 👏 
There are some issues that you still need to work on to go to the next project but you are almost there!



### Highlights 🎯 💡 
- Descriptive PR ✔️ 
- Professional README ✔️ 
- Clean code ✔️ 
- Linter checks ✔️ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/books/books.js (hunk 1 above)
Comment (isaacpitwa) [reply src/redux/books/books.js:22]: thanks, @codepantha, the action is being forced by eslint. 
it is for a default that will always be overridden by the values passed. it throws the error when I remove it. may be you can advise on another way to do it

File: src/redux/configureStore.js (hunk 2 above)
Comment (isaacpitwa) [reply src/redux/configureStore.js:2]: `CreateStore` is deprecated. attached is a screenshot of eslint validation.

Screenshot 2022-05-03 at 02 06 30

Comment (codepantha) [reply src/redux/configureStore.js:2]: You are right. Please go ahead and request another review 👍

File: src/redux/books/books.js (hunk 1 above)
Comment (codepantha) [reply src/redux/books/books.js:22]: @isaacpitwa if that is the case, then go ahead. I honestly have never come across it being done that way, but if it's being enforced directly by eslint, then you're probably good to go 😎

**Review by keroloussamy — APPROVED** [review]
Hi @isaacpitwa, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To highlight 🎯

- No linter errors.✔️
- Good code design and implementation. ✔️
- Add all the requirements. ✔️

clap



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**